src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs

```diff
@@ -121,7 +121,7 @@ namespace CommunityToolkit.Maui.Markup
 		/// </summary>
 		/// <param name=""textAlignmentControl""></param>
 		/// <returns>" + textAlignmentClass.Namespace + "." + textAlignmentClass.ClassName + @" with added <see cref=""TextAlignment.Start""/></returns>
-		public static " + textAlignmentClass.Namespace + "." + textAlignmentClass.ClassName + textAlignmentClass.GenericArguments + @" TextStart" + textAlignmentClass.GenericArguments + "(this " + textAlignmentClass.Namespace + "." + textAlignmentClass.ClassName + textAlignmentClass.GenericArguments + @" textAlignmentControl)" + textAlignmentClass.GenericConstraints + @"
+		public static TAssignable TextStart<TAssignable" + (string.IsNullOrWhiteSpace(textAlignmentClass.GenericArguments) ? ">" : ("," + textAlignmentClass.GenericArguments + ">")) + "(this TAssignable textAlignmentControl) where TAssignable : " + textAlignmentClass.Namespace + "." + textAlignmentClass.ClassName + (string.IsNullOrWhiteSpace(textAlignmentClass.GenericArguments) ? "" : "<") + textAlignmentClass.GenericArguments + (string.IsNullOrWhiteSpace(textAlignmentClass.GenericArguments) ? "" : ">") + textAlignmentClass.GenericConstraints + @"
```

Comment:
can we move this kind of check outside the template?
`(string.IsNullOrWhiteSpace(textAlignmentClass.GenericArguments)`

I'm suggesting it keep the code template easy to read

Reply:
Thanks!! Yea, I totally agree. 

I actually planned on moving that to its own method since I copy/pasted it about a dozen times in the Source Generator 😅 

I just wanted to get this PR submitted as a draft to get feedback on its results sooner rather than later since there may be multiple ways to fix #135, as demonstrated in #149.

Reply:
✅ Done